metadata/src/main/java/org/apache/kafka/controller/BrokerHeartbeatManager.java

```diff
@@ -82,23 +80,16 @@ static class BrokerHeartbeatState {
          */
         private long controlledShutdownOffset;
 
-        /**
-         * The previous entry in the unfenced list, or null if the broker is not in that list.
-         */
-        private BrokerHeartbeatState prev;
-
-        /**
-         * The next entry in the unfenced list, or null if the broker is not in that list.
-         */
-        private BrokerHeartbeatState next;
-
-        BrokerHeartbeatState(int id) {
+        BrokerHeartbeatState(
+            int id,
+            boolean fenced,
+            long metadataOffset,
+            long controlledShutdownOffset
```

Comment:
`controlledShutdownOffset` is an existing feature of the heartbeat manager, not added by this change. It is the offset at which the broker should complete its controlled shutdown. Used to prevent the broker shutting down before all the other brokers have fetched enough metadata.